File: Arcade/Intro/2 - Edge of the Ocean/7 - Almost Increasing Sequence/almostIncreasingSequence.java
boolean almostIncreasingSequence(int[] sequence) {

    int count = 0;
    
    for(int index = 0; index < sequence.length-1; index++)
    {
        boolean currentIsBiggerThanNext = sequence[index] >= sequence[index+1];
        boolean notAccessOutOfArray = index > 0 && index < sequence.length-2;
        
        if(notAccessOutOfArray) {
            boolean currentIsBiggerThanNextTwo = currentIsBiggerThanNext && sequence[index] >= sequence[index+2];
            boolean previousIsBiggerThanNext = sequence[index-1] >= sequence[index+1];
            
            if( currentIsBiggerThanNextTwo && previousIsBiggerThanNext) return false;
        }
        if(currentIsBiggerThanNext) count++;
        
        if(count > 1) return false;
    }
    
    if(count > 1) return false;
    return true;
}

File: Arcade/Intro/02 - Edge of the Ocean/7 - Almost Increasing Sequence/almostIncreasingSequence.cpp
bool almostIncreasingSequence(std::vector<int> sequence) {

    int count = 0;
    
    for(int index = 0; index < sequence.size()-1; index++)
    {
        bool currentIsBiggerThanNext = sequence.at(index) >= sequence.at(index+1);
        bool notAccessOutOfArray = index > 0 && index < sequence.size()-2;
        
        if ( notAccessOutOfArray ) {
            bool currentIsBiggerThanNextTwo = currentIsBiggerThanNext && sequence.at(index) >= sequence.at(index+2);
            bool previousIsBiggerThanNext = sequence.at(index-1) >= sequence.at(index+1);
            
            if(currentIsBiggerThanNextTwo && previousIsBiggerThanNext) {
                return false;
            }
        }
        
        if(currentIsBiggerThanNext) count++;
        
        if(count > 1) return false;
    }
    
    if(count > 1) return false;
    return true;
}
